### PR TITLE
:white_check_mark: Add features test and update the controllers

### PR DESCRIPTION
**Pull request by JrBour**
* Update swagger documentation
* Update controllers
* Add feature tests

### CODE REVIEW

**Comment by Sylcab74** [comment]
How to run tests, I have issues with sqlite database not found :cry:

File: .env.testing [hunk 1]
```diff
@@ -0,0 +1,39 @@
+APP_NAME=Laravel
+APP_ENV=local
+APP_KEY=base64:hRdl2Nk5p7X0IUhPlBJPmsm0xhJP+pQMai22zt/uroA=
+APP_DEBUG=true
+APP_URL=http://localhost
+LOG_CHANNEL=stack
+
+DB_CONNECTION=sqlite
+DB_DATABASE=/home/jrbour/Desktop/school/mentalist/database/database_testing.sqlite
```
Comment (Sylcab74) [inline .env.testing]: Dont use local path.

Comment (JrBour) [reply .env.testing]: I will add this file in the gitignore file

**Comment by JrBour** [comment]
> How to run tests, I have issues with sqlite database not found cry



> cry

Add a `.env.testing` file, add the route in the DB_CONNECTION variable for that laravel can access to your `.sqlite` file

File: tests/Feature/ArticleTest.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                'category_id',
+                'author_id'
+            ]);
+    }
+
+    public function testArticleUpdateFail()
+    {
+        $relationshipForUpdate = $this->createRelationshipObjectForArticle();
+        $data = [
+            'title' => 'RTFM',
```
Comment (Sylcab74) [inline tests/Feature/ArticleTest.php]: :rofl:

File: .env.testing (hunk 1 above)
Comment (Sylcab74) [reply .env.testing]: No it's ok, i fixed this

**Pull request merged**